src/app/upgrade.cpp

```diff
@@ -110,13 +111,121 @@ namespace
             settingsStorage->removeValue(oldKey);
         }
     }
+
+    void upgradeSchedulerDaysSettings()
+    {
+        auto *settingsStorage = SettingsStorage::instance();
+        const auto key = QString::fromLatin1("Preferences/Scheduler/days");
+        const auto value = settingsStorage->loadValue<QString>(key);
+
+        bool ok = false;
+        const auto number = value.toInt(&ok);
+
+        if (ok)
+        {
+            switch (number)
+            {
+            case 0:
+                settingsStorage->storeValue(key, Scheduler::Days::EveryDay);
+                break;
+            case 1:
+                settingsStorage->storeValue(key, Scheduler::Days::Weekday);
+                break;
+            case 2:
+                settingsStorage->storeValue(key, Scheduler::Days::Weekend);
+                break;
+            case 3:
+                settingsStorage->storeValue(key, Scheduler::Days::Monday);
+                break;
+            case 4:
+                settingsStorage->storeValue(key, Scheduler::Days::Tuesday);
+                break;
+            case 5:
+                settingsStorage->storeValue(key, Scheduler::Days::Wednesday);
+                break;
+            case 6:
+                settingsStorage->storeValue(key, Scheduler::Days::Thursday);
+                break;
+            case 7:
+                settingsStorage->storeValue(key, Scheduler::Days::Friday);
+                break;
+            case 8:
+                settingsStorage->storeValue(key, Scheduler::Days::Saturday);
+                break;
+            case 9:
+                settingsStorage->storeValue(key, Scheduler::Days::Sunday);
+                break;
+            default:
+                break;
```

Comment:
Why don't you "convert" invalid value to some "default" one? How application will behave when try to load enum from such invalid value?

Reply:
Seems I can delete the key and let `Preferences` class load the default value.

Reply:
> I should also add some logging to notify user if the logger is available...

Isn't this superfluous? Generally, this is a good idea. But it seems strange to me to have it here, but at the same time, during regular loading, leave incorrect values without attention.

Reply:
> I should also add some logging to notify user if the logger is available...

Isn't this superfluous? Generally, this is a good idea. But it seems strange to me to have it here, but at the same time, during regular loading, leave incorrect values without attention.